core/src/main/java/eu/fasten/core/data/ExtendedBuilder.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.fasten.core.data;
+
+/**
+ * Builder for {@link ExtendedRevisionCallGraph}.
+ */
+public abstract class ExtendedBuilder<A> {
+
+    protected String forge;
+    protected String product;
+    protected String version;
+    protected String cgGenerator;
+    protected long timestamp;
+    protected A classHierarchy;
+    protected Graph graph;
+    protected int nodeCount;
+
+    // TODO Do we need to convert this class to Singleton?
```

Comment:
@mir-am @MihhailSokolov Do we need to convert this class to Singleton?

Reply:
I don't think this is necessary. I try to avoid singletons in general unless it is really needed. But since this class is just a simple builder I think there is no need to make it a singleton.